src/allmydata/test/test_humanreadable.py

```diff
@@ -0,0 +1,55 @@
+"""Tests for allmydata.util.humanreadable."""
+from __future__ import unicode_literals
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from past.builtins import long
+
+from twisted.trial import unittest
+
+from allmydata.util import humanreadable
+
+
+
+def foo(): pass # keep the line number constant
+
+
+class NoArgumentException(Exception):
+    def __init__(self):
+        pass
+
+class HumanReadable(unittest.TestCase):
+    def test_repr(self):
+        hr = humanreadable.hr
+        self.failUnlessEqual(hr(foo), "<foo() at test_humanreadable.py:15>")
+        self.failUnlessEqual(hr(self.test_repr),
+                             "<bound method HumanReadable.test_repr of <allmydata.test.test_humanreadable.HumanReadable testMethod=test_repr>>")
+        self.failUnlessEqual(hr(long(1)), "1")
+        self.assertIn(hr(10**40),
+                      ["100000000000000000...000000000000000000",
+                       "100000000000000000...0000000000000000000"])
```

Comment:
... because of the trailing L on Python 2?  Nice ... Maybe it's not too hard to keep the behavior of the same?  Strip the L before doing the length math?  If it takes more than a minute maybe it's not worth it.

Reply:
It just gets shortened slightly differently, doesn't really seem meaningful.